jvm/core/src/main/scala/org/scalatest/tools/Framework.scala

```diff
@@ -337,8 +337,9 @@ class Framework extends SbtFramework {
         }
       }
     }
-    catch {       
+    catch {
       case e: Throwable => {
+        suiteSortingReporter.completedTests(suite.suiteId)
 
```

Comment:
I wonder if we call this again after the timer expires. It sounds like it may not matter with running with sbt, but I wonder if that causes issues when running not with sbt.

Reply:
Hey Bill!

`completedTests` method invokes `fireReadyEvents` which in turn calls `cancelTimeoutTask` to cancel the subject **timer**. Whether **timeout** task is thread-safe is a more difficult question which I'm afraid I can't answer. However, that could happen regardless of this PR's change I believe. Namely, a test suite completes in the happy code path and simulatenously **timeout** task gets fired after 2 sec.

Reply:
@bvenners @nimatrueway Framework.scala is called by sbt only, anyway, that also raises a question if we should do the similar when run with Runner, for Runner I think the similar spot is in SuiteRunner.scala: 

https://github.com/scalatest/scalatest/blob/main/jvm/core/src/main/scala/org/scalatest/tools/SuiteRunner.scala#L79

I am not sure how we can call completedTests there, we probably shouldn't.  Different from the sbt runner, the ScalaTest Runner will wait until the timeout and got the SuiteAborted eventually, I think.